MIGRATING.md

```diff
@@ -0,0 +1,71 @@
+# Okta Java Management SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's [library version policy](https://developer.okta.com/code/library-versions/). In short, we don't make breaking changes unless the major version changes!
+
+## Migrating from 1.x.x to 2.0.0
+
+Version 2.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing class/interfaces are no longer backward compatible due to method renaming and signature changes, see [Breaking Changes](#breaking-changes).
+
+### Breaking Changes
+
+Following is a list of changes that break backward compatibility in version 2.0.0.
+
+- `com.okta.sdk.client.Client`
+   - Renamed `createRule` to `createGroupRule`
+   - Renamed `getRule` to `getGroupRule`
+   - Renamed `listRules` to `listGroupRules`
+   
+- `com.okta.sdk.resource.log.LogEventList` 
+   - `getLogs(String until, String since, String filter, String q, String sortOrder)`
+   Signature changed to `getLogs(Date since, Date until, String filter, String q, String sortOrder)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(String q, String filter, String expand)`
+   Signature changed to `listGroups(String q, String filter)`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+   
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - `delete(Boolean removeUsers)` Signature changed to `delete()`
+   - Removed `getAllGroupsValid` and `setAllGroupsValid` (property `allGroupsValid` is removed)
```

Comment:
we may need to give context to the reader, was this value always null? why was it removed? Could I have previously used that value?

Reply:
Looked at https://developer.okta.com/docs/reference/api/groups/#group-attributes for property `allGroupsValid` and I don't seem to find it. I am not sure how it might have been used in the past but it is present in our current spec (1.6.0).
We removed it now in v2.0.0 spec.

Reply:
@bretterer do you have any background on this change?

Reply:
I am not sure of past use of these methods.

Reply:
@bdemers @bretterer recommendations on how to word this change here?

Reply:
I'm not sure what the purpose of this property was or, how it could be used.  But even saying something like, this option is being removed in future versions of the Okta API would be fine, (ideally with an explanation, but 🤷 )

Reply:
updated